### ISSUE TITLE
IgnoreBlankLines behavior change

### ISSUE THREAD

**Issue by ArdentLogician** [JoshClose/CsvHelper#926]
I'm currently in the process of upgrading a set of parsing libraries from using 2.16.3 to the latest version (6.1.1 as of now). The files being parsed are not strictly CSV, but a mixed bag of formats where CSV covers the majority of it, and so the additional aspects are handled using various facilities of the package such as looking directly at the ReadingContext.RawRecord (formerly Parser.RawRecord) and doing string comparisons. There is a change in behavior when (Csv)Configuration.IgnoreBlankLines = false that is breaking all of these parsers as a result. Fixing it is unlikely to be difficult, but the new behavior seems unusual at best. When reading:

Line 1: "'Some','Stuff','On','Line'\r\n"
Line 2: "\r\n"
Line 3: "'Next','Line'\r\n"

In 2.16.3 the RawRecord returns (on consecutive calls to Read):
Line 1: "'Some','Stuff','On','Line'\r\n"
Line 2: "\r\n"
Line 3: "'Next','Line'\r\n"

In 6.1.1 the RawRecord returns (on consecutive calls to Read):
Line 1: "'Some','Stuff','On','Line'\r\n"
Line 2: ""
Line 3: "\r\n'Next','Line'\r\n"

In other words, the new version is carrying the previous CRLF from the ignored blank line onto the RawRecord of line 3. In 2.16.3, this CRLF was properly (IMHO) assigned to the skipped line. This results in many of our comparisons to RawRecord failing because they are using String.StartsWith(....)

Obviously this can be fixed in the comparison in a number of ways, but it seems nonsensical to have to do this as the offending characters do not belong to that line.

Is this a deliberate change? An oversight? A bug?

**Comment by JoshClose** [comment]
I would say this is a bug. I'll see what I can do to fix it.

**Comment by JoshClose** [comment]
After looking into this, I don't see how this can be avoided. When `IgnoreBlankLines` is on, you don't get that row from calling `Read`. The only way to get the actual raw data would be to then give it on the next call.

If you can think of a better way, I'm open to suggestions.
